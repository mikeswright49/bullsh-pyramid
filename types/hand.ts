import { Card } from './card';

export type Hand = {
    players: Card[][];
    tiers: Card[][];
};
