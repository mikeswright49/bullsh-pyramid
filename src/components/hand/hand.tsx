import { Card } from '../card/card';
import styles from './hand.module.css';
import React from 'react';
import { Card as CardType } from '../../../types/card';

export interface HandProps {
    cards: CardType[];
    showSelector: boolean;
}
export function Hand({ cards, showSelector }: HandProps): JSX.Element {
    if (!cards) {
        return null;
    }

    return (
        <>
            <h3>Your hand</h3>
            <div className={styles.hand}>
                {cards.map((card) => (
                    <div key={`hand-card-${card.value}-${card.suit}`}>
                        <Card card={card} />
                        {showSelector && <button>This one</button>}
                    </div>
                ))}
            </div>
        </>
    );
}
