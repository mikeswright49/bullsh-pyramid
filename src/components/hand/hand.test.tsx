import { render } from '@testing-library/react';
import React from 'react';
import { Hand, HandProps } from './hand';

describe('<Unit Test> Hand', () => {
    let mockProps: HandProps;
    beforeEach(() => {
        mockProps = {
            showSelector: false,
            cards: [
                {
                    value: 1,
                    hidden: false,
                    suit: 0,
                },
                {
                    value: 2,
                    hidden: false,
                    suit: 0,
                },
                {
                    value: 3,
                    hidden: false,
                    suit: 0,
                },
            ],
        };
    });

    it('should be able to render the component', () => {
        const { container } = render(<Hand {...mockProps} />);
        expect(container).toMatchSnapshot();
    });

    it('should be able to show the selection button', () => {
        mockProps.showSelector = true;
        const { container } = render(<Hand {...mockProps} />);
        expect(container).toMatchSnapshot();
    });
});
