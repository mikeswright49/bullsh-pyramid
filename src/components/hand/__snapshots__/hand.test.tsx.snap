// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Unit Test> Hand should be able to render the component 1`] = `
<div>
  <h3>
    Your hand
  </h3>
  <div
    class="hand"
  >
    <div>
      <div
        class="flex-center-center card"
        style="background-position: -79px -0px;"
      />
    </div>
    <div>
      <div
        class="flex-center-center card"
        style="background-position: -158px -0px;"
      />
    </div>
    <div>
      <div
        class="flex-center-center card"
        style="background-position: -237px -0px;"
      />
    </div>
  </div>
</div>
`;

exports[`<Unit Test> Hand should be able to show the selection button 1`] = `
<div>
  <h3>
    Your hand
  </h3>
  <div
    class="hand"
  >
    <div>
      <div
        class="flex-center-center card"
        style="background-position: -79px -0px;"
      />
      <button>
        This one
      </button>
    </div>
    <div>
      <div
        class="flex-center-center card"
        style="background-position: -158px -0px;"
      />
      <button>
        This one
      </button>
    </div>
    <div>
      <div
        class="flex-center-center card"
        style="background-position: -237px -0px;"
      />
      <button>
        This one
      </button>
    </div>
  </div>
</div>
`;
